File: app/src/main/java/ru/fitsme/android/presentation/mapper/Mapper.java
package ru.fitsme.android.presentation.mapper;

/**
 * Interface for model mappers. It provides helper methods that facilitate
 * retrieving of models from outer layers
 *
 * @param <D> the input type (base model)
 * @param <V> the view model output type
 */
public interface Mapper<V, D> {

    V mapToViewModel(D type);

    D mapFromViewModel(V type);

}

File: app/src/main/java/ru/fitsme/android/presentation/mapper/Mapper.kt
package ru.fitsme.android.presentation.mapper

/**
 * Interface for model mappers. It provides helper methods that facilitate
 * retrieving of models from outer layers
 *
 * @param <V> the input type (base model)
 * @param <D> the view model output type
 */
interface Mapper<out V, in D> {

    fun mapToViewModel(type: D): V

}